app/src/main/java/org/hackillinois/android/view/SplashScreenActivity.kt

```diff
@@ -25,32 +46,41 @@ class SplashScreenActivity : AppCompatActivity() {
             api.user().enqueue(object : Callback<User> {
                 override fun onFailure(call: Call<User>, t: Throwable) {
                     Log.e("LoginActivity", "Failed to check is jwt is valid")
-                    runOnUiThread {
-                        launchLoginActivity()
-                    }
+                    needsToLogin = true
+                    countDownLatch.countDown()
                 }
 
                 override fun onResponse(call: Call<User>, response: Response<User>) {
-                    if (response.code() == 200) {
+                    needsToLogin = if (response.code() == 200) {
```

Comment:
You could do `isLoggedIn = (response.code() == 200)` right off the bat. Then use `isLoggedIn` in the `if`